tt_metal/CMakeLists.txt

```diff
@@ -31,6 +31,7 @@ target_link_libraries(
         magic_enum
         fmt
         span
+        Boost::container
```

Comment:
Would appreciate if you follow the pattern of the interface library I did for `span`, like:
```cmake
add_library(small_vector INTERFACE)
target_link_libraries(small_vector INTERFACE Boost::container)
```
Then link against the `small_vector` library instead of `Boost::container` directly, which will reduce friction if we move away from boost in the future, which is still under a lot of discussion right now. Also documents in CMake exactly _why_ we pull in that dependency.

Reply:
@patrickroberts I'm quite unsure what this would achieve. 
If we add `add_library(small_vector INTERFACE)` with `small_vector.hpp` specified as a source, it would bring more clarity for what's going on, but unfortunately we can't do this, because `small_vector.hpp` depends on other parts of ttnn.
On the other hand, if we just create an empty interface library, it would be just misleading, because the code `target_link_libraries(ttnn PUBLIC small_vector)` would actually import whole `Boost::container`.

Reply:
What it achieves is that if we swap out the implementation of `SmallVector` for something else besides `Boost::container`, we don't have to guess exactly what depends on `Boost::container` for that specific purpose, search all the CMake files that have that dependency, and replace it with something else. There would be only one place in CMake to change in order to move to a different dependency.

Reply:
Roughly the same thing goes about a comment. The correct way to fina all usages of Boost::containter is to do a search by `#include <boost/container`. If I leave a comment that I need Boost::container for small_vector, it's likely that someone will just use one of the containers from boost without updating the comment, making it misguiding.

Reply:
You've already put in the work of implementing a proper wrapper container so that nobody should be directly including `<boost/container` anyway. I don't want to go down the path of defending one bad practice with the possibility of another.

Reply:
You say "search all the CMake files that have that dependency", but this can't be done. Because we can't do `add_library(small_vector INTERFACE small_vector.hpp)`, because `small_vector.hpp` depends on a bunch of stuff in ttnn.

And again, even if we could do it, wouldn't help in case we decide to replace boost. For example `free_list.cpp` actually depends on `Boost::smart_ptr` which isn't shown anywhere in cmake.

Reply:
Hiding boost behind an interface specifying the clear reason why it is brought up sounds like a great idea.
But it actually does not prevent developers from pulling in other types from boost::containers after that interface library is linked. What starts with a neat intention will quickly start lying to us.

Please don't do this.

Reply:
Evidently I didn't look closely enough on my first pass.  The only change in `tt_metal` that I see is a new usage of `std::span`.

Two questions:
* What's actully using Boost?  I don't see `boost::span`, only `std::span`
* Is `common/test_tiles.hpp` part of the public API?  (It's not clear to me from the directory layout)

read: it's not clear to me why the new dependency is (a) Boost and (b) public.

Reply:
boost::small_vector

Reply:
Oh, actually after a second look I agree. It is only used in ttnn right now. 
@sminakov-tt I think it can be removed from metal cmake, wdyt?

Reply:
@ayerofieiev-tt Yes, I definitely wanted to include it only in `ttnn`, not for the whole `tt_metal`. Fixed the cmake files.